### ISSUE TITLE
[FR] Make the sun great again

### ISSUE THREAD

**Issue by sprunk** [beyond-all-reason/spring#871]
Some context: https://github.com/beyond-all-reason/spring/pull/833

Currently there is no simple (one-two callouts that just take a number) way to say "the sun is now less bright".

There is this huge table of parameters (diffuse color, specular color, shadow density, etc): https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/luarules/gadgets/map_nightmode.lua#L129-L154

And to tweak the sun (less bright, different hue) you need to apply the change to them all: https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/luarules/gadgets/map_nightmode.lua#L287-L300

There is some sort of `intensity` parameter to `Spring.SetSunDirection` which currently doesn't do anything, and is also misplaced (if anything, it belongs with sun color).

I think ideally you would set stuff like ambient light level once, and then `sunColor` (float3) and `intensity` (float) would multiply all the other parameters automatically. So for example:
 * if `intensity` is 0.468 then `shadowDensity` also automatically becomes 0.468 (currently you need to set shadow density explicitly). Perhaps you can set some "base" shadow density or opt out of the automated system entirely, but the defaults would ideally work automagically.
 * if ambient color of ground (or units) is `{0.2, 0.4, 0.6}` and `intensity` is 0.5 then the actual color of units automatically becomes `{0.6, 0.7, 0.8}` (currently you need to set the color explicitly, and separately from shadow density).
 * in the above example, if `sunColor` is then changed to `{0, 1, 1}` then the actual color of units automatically becomes `{0.2, 0.7, 0.8}`. Perhaps `sunColor` and `intensity` should be a single float4, after all intensity is a lot like alpha, though it makes sense the current way too.
* specular is adjusted as well (currently also yet another separate entry) as well as anything else that might be missing in the BAR widget above (it's probably comprehensive, but it's burdensome to keep track of all the parameters) or that might be added in the future.

That's all ground for a larger rework though, and fairly low priority.

**Comment by Beherith** [comment]
You dont _need_ to apply them all, only the ones that you change. But you need to apply them twice ;)

**Comment by sprunk** [comment]
> You dont need to apply them all, only the ones that you change.

Yes, but that is part of my complaint. Having them changeable separately is unintuitive. If you set `shadowDensity` to `0` then the sun isn't shining, right? But then if you set `unitDiffuseColor` higher than `unitAmbientColor` then something has to be casting light on them after all. And then you can set `sunColor` to `1 0 0` red but `unitDiffuseColor` to `0 0 1` blue on top of that for a very confusing lighting. But if the sun is glowing pure red, then units can only be as blue as the ambient light allows. And if the sun is not casting shadows then how is anything receiving non-ambient light?

Essentially, unless you're going for funky lighting that has no basis in physics, you'll want to change and apply most of the parameters anyway, since most of them depend on the intensity of light. Having to apply them twice sounds like a bug but is otherwise orthogonal to this request.
